cmd/dep/status.go

```diff
@@ -150,6 +158,123 @@ func (out *jsonOutput) MissingFooter() {
 	json.NewEncoder(out.w).Encode(out.missing)
 }
 
+type dotProject struct {
+	project  string
+	version  string
+	children []string
+}
+
+func (dp *dotProject) hash() uint32 {
+	h := fnv.New32a()
+	h.Write([]byte(dp.project))
+	return h.Sum32()
+}
+
+func (dp *dotProject) label() string {
+	label := []string{dp.project}
+
+	if dp.version != "" {
+		label = append(label, dp.version)
+	}
+
+	return strings.Join(label, "\n")
+}
+
+type dotOutput struct {
+	w   io.Writer
+	o   string
+	p   *dep.Project
+	dps []*dotProject
+	b   bytes.Buffer
+	bsh map[string]uint32
+}
+
+func (out *dotOutput) BasicHeader() {
+
+	// TODO Check for dot package before doing something
+	// cmd := exec.Command("dot", "-V")
+
+	out.dps = []*dotProject{}
+	out.bsh = make(map[string]uint32)
+	out.b.WriteString("digraph { node [shape=box]; ")
+
+	ptree, _ := gps.ListPackages(out.p.AbsRoot, string(out.p.ImportRoot))
+	prm, _ := ptree.ToReachMap(true, false, false, nil)
+
+	rdp := &dotProject{
+		project:  string(out.p.ImportRoot),
+		version:  "",
+		children: prm.Flatten(false),
+	}
+
+	out.dps = append(out.dps, rdp)
+}
+
+func (out *dotOutput) BasicFooter() {
+
+	for _, dp := range out.dps {
+		out.bsh[dp.project] = dp.hash()
+
+		// Create name boxes, and name them using hashes
+		// to avoid encoding name conflicts
+		out.b.WriteString(fmt.Sprintf("%d [label=\"%s\"];", dp.hash(), dp.label()))
+	}
+
+	// Store relations to avoid duplication
+	rels := make(map[string]bool)
+
+	// Create relations
+	for _, dp := range out.dps {
+		for _, bsc := range dp.children {
+			for pr, hsh := range out.bsh {
+				if strings.HasPrefix(bsc, pr) {
+					r := fmt.Sprintf("%d -> %d", out.bsh[dp.project], hsh)
+
+					if _, ex := rels[r]; !ex {
+						out.b.WriteString(r + "; ")
+						rels[r] = true
+					}
+
+				}
+			}
+		}
+	}
+
+	out.b.WriteString("}")
+
+	// TODO: Pipe created string to dot
+	// Storing Graphviz output inside temp file to generate dot output
+	tf, err := ioutil.TempFile(os.TempDir(), "")
+
+	if err != nil {
+		fmt.Printf("%v", err)
+	}
+
+	defer syscall.Unlink(tf.Name())
```

Comment:
This should all go away when switching to just printing to stdout instead of a tmp file, but for future - `os.Remove` is the call to make here.